CONTRIBUTING.md

```diff
@@ -44,23 +47,21 @@ Examples of bad commit messages:
 
 # Code style
 
-We use [Microsoft C# Coding Conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions)
-throughout the repository, with several exceptions:
+We use [Microsoft C# Coding Conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions) throughout the repository, with several exceptions:
```

Comment:
consider updating the link (https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)

there's also a few more places were we break those guidelines, such as:
- banning `var`
- not using ConfigureAwait
- using collection expressions (i assume these guidelines just haven't been updated)
- using LINQ method syntax, or not using it at all
- using file scoped namespaces
- that document is telling you to limit file length but you're changing that in this very PR